Application.mk

```diff
@@ -69,6 +69,18 @@ ifneq ($(CONFIG_BUILD_KERNEL),y)
   endif
 endif
 
+# When building a module, link with the compiler runtime.
+# This should be linked after libapps. Consider that mbedtls in libapps
+# uses __udivdi3.
+ifeq ($(BUILD_MODULE),y)
+  # Revisit: This only works for gcc and clang.
+  # Do other compilers have similar?
+  COMPILER_RT_LIB = $(shell $(CC) $(ARCHCPUFLAGS) --print-libgcc-file-name)
+  ifneq ($(COMPILER_RT_LIB),)
+    LDLIBS += $(COMPILER_RT_LIB)
```

Comment:
should we add EXTRA_LIB to the link in case of build module?

Reply:
iirc EXTRA_LIBS is to build the nuttx image and contains other things like proprietary drivers for esp32.

Reply:
maybe we can make make.defs do
```
COMPILER_RT_LIBS += libgcc
EXTRA_LIBS += $(COMPILER_RT_LIBS)
```
instead of
```
EXTRA_LIBS += libgcc
```

how do you think?

Reply:
What's different?

Reply:
COMPILER_RT_LIBS would not have things like proprietary drivers.

Reply:
I am fine with either.